General code cleanup, including threadsafety. (#29)
logstash-plugins/logstash-filter-cipher: docs/index.asciidoc

  * Value type is <<string,string>>
  * There is no default value for this setting.

The cipher algorithm to use for encryption and decryption operations.

> A list of supported algorithms depends on the versions of Logstash, JRuby, and Java this plugin is running in, but can be obtained by

Reviewer: ```suggestion
A list of supported algorithms depends on the versions of Logstash, JRuby, and Java this plugin is running in, but can be obtained by running:
```

> A list of supported algorithms depends on the versions of Logstash, JRuby, and Java this plugin is running in, but can be obtained by
> [source,ruby]
>     puts OpenSSL::Cipher.ciphers

Author: @karenzone this combines your input, and is also even more helpful because it references a command they can actually run:

```suggestion
A list of supported algorithms depends on the versions of Logstash, JRuby, and Java this plugin is running in, but can be obtained by running:
[source,sh]
    cd $LOGSTASH_HOME # <-- your Logstash distribution root
    bin/ruby -ropenssl -e 'puts OpenSSL::Cipher.ciphers'
```

  * Value type is <<number,number>>
  * There is no default value for this setting.

In encryption operations, this plugin generates a random Initialization

> Vector (IV) per encryption operation, a standard best-practice to ensure

Reviewer: ```suggestion
Vector (IV) per encryption operation. This is a standard best-practice to ensure
```

Reviewer: Note: It's not incorrect the way it is. Shorter sentences are easier for non-native speakers to parse and easier to translate.

  * Value type is <<string,string>>
  * There is no default value for this setting.

> The key to use for encryption and decryption operations

Reviewer: ```suggestion
The key to use for encryption and decryption operations.
```

  * Value type is <<string,string>>
  * Default value is `"\u0000"`

> The character used to pad the key to the required <<plugins-{type}s-{plugin}-key_size>>

Reviewer: ```suggestion
The character used to pad the key to the required <<plugins-{type}s-{plugin}-key_size>>.
```

  * Default value is `1`

> If this is set the internal Cipher instance will be

Reviewer: ```suggestion
If this value is set, the internal Cipher instance will be
```

> re-used up to `max_cipher_reuse` times before re-created

Reviewer: ```suggestion
re-used up to `max_cipher_reuse` times before it is re-created
```

  * This is a required setting.
  * Value type is <<string,string>>

>   ** `encrypt`: perform encryption of a plaintext value into IV + ciphertext
>   ** `decrypt`: perform decryption of an IV + ciphertext value into plaintext

Reviewer: ```suggestion
  ** `encrypt`: encrypts a plaintext value into IV + ciphertext
  ** `decrypt`: decrypts an IV + ciphertext value into plaintext
```

  * There is no default value for this setting.

[id="plugins-{type}s-{plugin}-source"]
===== `source` 

  * Value type is <<string,string>>
  * Default value is `"message"`

> The name of the source field

Reviewer: ```suggestion
The name of the source field.
```
